### ISSUE TITLE
Flat mode fail to show image

### ISSUE THREAD

**Issue by martadinata666** [bpatrik/pigallery2#846]
**Note:** Please always try reproducing the issue on the `:edge` build. There is a chance that it was already fixed. If it's an UI issue, you can try reporducing it on the [demo page](https://pigallery2.onrender.com/). 

----------------------------------------------------------------------------------

**Describe the bug**
Flat directory mode error

**Photo/video (optional) that causes the bug**
![2024-03-07-092856_1900x267_scrot](https://github.com/bpatrik/pigallery2/assets/2433562/b47417dc-a5d9-44e2-8c82-09944f1fdf7b)

**Server logs (optional)**
```
pigallery2_app.1.2hakjed8hc4y@homelab    | ErrorDTO {
pigallery2_app.1.2hakjed8hc4y@homelab    |   code: 31,
pigallery2_app.1.2hakjed8hc4y@homelab    |   message: 'Error during searching',
pigallery2_app.1.2hakjed8hc4y@homelab    |   details: QueryFailedError: ER_BAD_FIELD_ERROR: Unknown column 'media.metadataCreationdateoffset' in 'field list'
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Query.<anonymous> (/pigallery2/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:158:37)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Query.<anonymous> (/pigallery2/node_modules/mysql/lib/Connection.js:526:10)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Query._callback (/pigallery2/node_modules/mysql/lib/Connection.js:488:16)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Sequence.end (/pigallery2/node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Query.ErrorPacket (/pigallery2/node_modules/mysql/lib/protocol/sequences/Query.js:92:8)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Protocol._parsePacket (/pigallery2/node_modules/mysql/lib/protocol/Protocol.js:291:23)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Parser._parsePacket (/pigallery2/node_modules/mysql/lib/protocol/Parser.js:433:10)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Parser.write (/pigallery2/node_modules/mysql/lib/protocol/Parser.js:43:10)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Protocol.write (/pigallery2/node_modules/mysql/lib/protocol/Protocol.js:38:16)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Socket.<anonymous> (/pigallery2/node_modules/mysql/lib/Connection.js:88:28)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Socket.<anonymous> (/pigallery2/node_modules/mysql/lib/Connection.js:526:10)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Socket.emit (node:events:518:28)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at addChunk (node:internal/streams/readable:559:12)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at Readable.push (node:internal/streams/readable:390:5)
pigallery2_app.1.2hakjed8hc4y@homelab    |       at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
pigallery2_app.1.2hakjed8hc4y@homelab    |     query: 'SELECT `media`.`id` AS `media_id`, `media`.`name` AS `media_name`, `media`.`type` AS `media_type`, `media`.`directoryId` AS `media_directoryId`, `media`.`metadataCaption` AS `media_metadataCaption`, `media`.`metadataTitle` AS `media_metadataTitle`, `media`.`metadataCreationdate` AS `media_metadataCreationdate`, `media`.`metadataCreationdateoffset` AS `media_metadataCreationdateoffset`, `media`.`metadataFilesize` AS `media_metadataFilesize`, `media`.`metadataKeywords` AS `media_metadataKeywords`, `media`.`metadataRating` AS `media_metadataRating`, `media`.`metadataFaces` AS `media_metadataFaces`, `media`.`metadataBitrate` AS `media_metadataBitrate`, `media`.`metadataDuration` AS `media_metadataDuration`, `media`.`metadataSizeWidth` AS `media_metadataSizeWidth`, `media`.`metadataSizeHeight` AS `media_metadataSizeHeight`, `media`.`metadataCameraDataIso` AS `media_metadataCameraDataIso`, `media`.`metadataCameraDataModel` AS `media_metadataCameraDataModel`, `media`.`metadataCameraDataMake` AS `media_metadataCameraDataMake`, `media`.`metadataCameraDataFstop` AS `media_metadataCameraDataFstop`, `media`.`metadataCameraDataExposure` AS `media_metadataCameraDataExposure`, `media`.`metadataCameraDataFocallength` AS `media_metadataCameraDataFocallength`, `media`.`metadataCameraDataLens` AS `media_metadataCameraDataLens`, `media`.`metadataPositionDataCountry` AS `media_metadataPositionDataCountry`, `media`.`metadataPositionDataState` AS `media_metadataPositionDataState`, `media`.`metadataPositionDataCity` AS `media_metadataPositionDataCity`, `media`.`metadataPositionDataGPSDataLatitude` AS `media_metadataPositionDataGPSDataLatitude`, `media`.`metadataPositionDataGPSDataLongitude` AS `media_metadataPositionDataGPSDataLongitude`, `directory`.`id` AS `directory_id`, `directory`.`name` AS `directory_name`, `directory`.`path` AS `directory_path` FROM `media_entity` `media` LEFT JOIN `directory_entity` `directory` ON `directory`.`id`=`media`.`directoryId` WHERE (`directory`.`path` LIKE ? COLLATE utf8mb4_general_ci OR (`directory`.`name` LIKE ? COLLATE utf8mb4_general_ci)) LIMIT 10001',
pigallery2_app.1.2hakjed8hc4y@homelab    |     parameters: [ '%.%', '%.%' ],
pigallery2_app.1.2hakjed8hc4y@homelab    |     driverError: Error: ER_BAD_FIELD_ERROR: Unknown column 'media.metadataCreationdateoffset' in 'field list'
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Sequence._packetToError (/pigallery2/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Query.ErrorPacket (/pigallery2/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Protocol._parsePacket (/pigallery2/node_modules/mysql/lib/protocol/Protocol.js:291:23)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Parser._parsePacket (/pigallery2/node_modules/mysql/lib/protocol/Parser.js:433:10)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Parser.write (/pigallery2/node_modules/mysql/lib/protocol/Parser.js:43:10)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Protocol.write (/pigallery2/node_modules/mysql/lib/protocol/Protocol.js:38:16)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Socket.<anonymous> (/pigallery2/node_modules/mysql/lib/Connection.js:88:28)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Socket.<anonymous> (/pigallery2/node_modules/mysql/lib/Connection.js:526:10)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Socket.emit (node:events:518:28)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at addChunk (node:internal/streams/readable:559:12)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Readable.push (node:internal/streams/readable:390:5)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
pigallery2_app.1.2hakjed8hc4y@homelab    |         --------------------
pigallery2_app.1.2hakjed8hc4y@homelab    |         at Protocol._enqueue (/pigallery2/node_modules/mysql/lib/protocol/Protocol.js:144:48)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at PoolConnection.query (/pigallery2/node_modules/mysql/lib/Connection.js:198:25)
pigallery2_app.1.2hakjed8hc4y@homelab    |         at /pigallery2/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:148:36
pigallery2_app.1.2hakjed8hc4y@homelab    |         at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
pigallery2_app.1.2hakjed8hc4y@homelab    |       code: 'ER_BAD_FIELD_ERROR',
pigallery2_app.1.2hakjed8hc4y@homelab    |       errno: 1054,
pigallery2_app.1.2hakjed8hc4y@homelab    |       sqlMessage: "Unknown column 'media.metadataCreationdateoffset' in 'field list'",
pigallery2_app.1.2hakjed8hc4y@homelab    |       sqlState: '42S22',
pigallery2_app.1.2hakjed8hc4y@homelab    |       index: 0,
pigallery2_app.1.2hakjed8hc4y@homelab    |       sql: "SELECT `media`.`id` AS `media_id`, `media`.`name` AS `media_name`, `media`.`type` AS `media_type`, `media`.`directoryId` AS `media_directoryId`, `media`.`metadataCaption` AS `media_metadataCaption`, `media`.`metadataTitle` AS `media_metadataTitle`, `media`.`metadataCreationdate` AS `media_metadataCreationdate`, `media`.`metadataCreationdateoffset` AS `media_metadataCreationdateoffset`, `media`.`metadataFilesize` AS `media_metadataFilesize`, `media`.`metadataKeywords` AS `media_metadataKeywords`, `media`.`metadataRating` AS `media_metadataRating`, `media`.`metadataFaces` AS `media_metadataFaces`, `media`.`metadataBitrate` AS `media_metadataBitrate`, `media`.`metadataDuration` AS `media_metadataDuration`, `media`.`metadataSizeWidth` AS `media_metadataSizeWidth`, `media`.`metadataSizeHeight` AS `media_metadataSizeHeight`, `media`.`metadataCameraDataIso` AS `media_metadataCameraDataIso`, `media`.`metadataCameraDataModel` AS `media_metadataCameraDataModel`, `media`.`metadataCameraDataMake` AS `media_metadataCameraDataMake`, `media`.`metadataCameraDataFstop` AS `media_metadataCameraDataFstop`, `media`.`metadataCameraDataExposure` AS `media_metadataCameraDataExposure`, `media`.`metadataCameraDataFocallength` AS `media_metadataCameraDataFocallength`, `media`.`metadataCameraDataLens` AS `media_metadataCameraDataLens`, `media`.`metadataPositionDataCountry` AS `media_metadataPositionDataCountry`, `media`.`metadataPositionDataState` AS `media_metadataPositionDataState`, `media`.`metadataPositionDataCity` AS `media_metadataPositionDataCity`, `media`.`metadataPositionDataGPSDataLatitude` AS `media_metadataPositionDataGPSDataLatitude`, `media`.`metadataPositionDataGPSDataLongitude` AS `media_metadataPositionDataGPSDataLongitude`, `directory`.`id` AS `directory_id`, `directory`.`name` AS `directory_name`, `directory`.`path` AS `directory_path` FROM `media_entity` `media` LEFT JOIN `directory_entity` `directory` ON `directory`.`id`=`media`.`directoryId` WHERE (`directory`.`path` LIKE '%.%' COLLATE utf8mb4_general_ci OR (`directory`.`name` LIKE '%.%' COLLATE utf8mb4_general_ci)) LIMIT 10001"
pigallery2_app.1.2hakjed8hc4y@homelab    |     },
pigallery2_app.1.2hakjed8hc4y@homelab    |     code: 'ER_BAD_FIELD_ERROR',
pigallery2_app.1.2hakjed8hc4y@homelab    |     errno: 1054,
pigallery2_app.1.2hakjed8hc4y@homelab    |     sqlMessage: "Unknown column 'media.metadataCreationdateoffset' in 'field list'",
pigallery2_app.1.2hakjed8hc4y@homelab    |     sqlState: '42S22',
pigallery2_app.1.2hakjed8hc4y@homelab    |     index: 0,
pigallery2_app.1.2hakjed8hc4y@homelab    |     sql: "SELECT `media`.`id` AS `media_id`, `media`.`name` AS `media_name`, `media`.`type` AS `media_type`, `media`.`directoryId` AS `media_directoryId`, `media`.`metadataCaption` AS `media_metadataCaption`, `media`.`metadataTitle` AS `media_metadataTitle`, `media`.`metadataCreationdate` AS `media_metadataCreationdate`, `media`.`metadataCreationdateoffset` AS `media_metadataCreationdateoffset`, `media`.`metadataFilesize` AS `media_metadataFilesize`, `media`.`metadataKeywords` AS `media_metadataKeywords`, `media`.`metadataRating` AS `media_metadataRating`, `media`.`metadataFaces` AS `media_metadataFaces`, `media`.`metadataBitrate` AS `media_metadataBitrate`, `media`.`metadataDuration` AS `media_metadataDuration`, `media`.`metadataSizeWidth` AS `media_metadataSizeWidth`, `media`.`metadataSizeHeight` AS `media_metadataSizeHeight`, `media`.`metadataCameraDataIso` AS `media_metadataCameraDataIso`, `media`.`metadataCameraDataModel` AS `media_metadataCameraDataModel`, `media`.`metadataCameraDataMake` AS `media_metadataCameraDataMake`, `media`.`metadataCameraDataFstop` AS `media_metadataCameraDataFstop`, `media`.`metadataCameraDataExposure` AS `media_metadataCameraDataExposure`, `media`.`metadataCameraDataFocallength` AS `media_metadataCameraDataFocallength`, `media`.`metadataCameraDataLens` AS `media_metadataCameraDataLens`, `media`.`metadataPositionDataCountry` AS `media_metadataPositionDataCountry`, `media`.`metadataPositionDataState` AS `media_metadataPositionDataState`, `media`.`metadataPositionDataCity` AS `media_metadataPositionDataCity`, `media`.`metadataPositionDataGPSDataLatitude` AS `media_metadataPositionDataGPSDataLatitude`, `media`.`metadataPositionDataGPSDataLongitude` AS `media_metadataPositionDataGPSDataLongitude`, `directory`.`id` AS `directory_id`, `directory`.`name` AS `directory_name`, `directory`.`path` AS `directory_path` FROM `media_entity` `media` LEFT JOIN `directory_entity` `directory` ON `directory`.`id`=`media`.`directoryId` WHERE (`directory`.`path` LIKE '%.%' COLLATE utf8mb4_general_ci OR (`directory`.`name` LIKE '%.%' COLLATE utf8mb4_general_ci)) LIMIT 10001"
pigallery2_app.1.2hakjed8hc4y@homelab    |   },
pigallery2_app.1.2hakjed8hc4y@homelab    |   request: { method: '', url: '' }
pigallery2_app.1.2hakjed8hc4y@homelab    | }
pigallery2_app.1.2hakjed8hc4y@homelab    | 3/7/2024, 9:23:07 AM[VERBS] GET / 200 135ms

```

** Questions?
Do we need database reset? As this likely missing column. The deployment is upgrade, from previous installation build on 27/02/2024.

In case of a server side error, provide logs:
* Set env variable `NODE_ENV=debug` to log everything.
* To get logs from docker: `docker ps -a` to list containers, `docker logs <container ID>` to get logs.

**Environment (please complete the following information):**
 - OS: linux
 - Browser chrome

**Used app version**:
- compiled from source 7/3/2024

**Comment by martadinata666** [comment]
TLDR; Database reset needed, maybe also consider adding the missing column automatically?

**Comment by bpatrik** [comment]
@grasdk can you bump the database version number in the common folder? (On my phone so I can't point to the code now)

That would trigger a DB reset on startup.

Adding new collums to the DB requires that as there are no reliable update scripts.

**Comment by grasdk** [comment]
@bpatrik PR ready: https://github.com/bpatrik/pigallery2/pull/851
